crates/uv-resolver/src/marker.rs

```diff
@@ -35,7 +35,8 @@ pub(crate) fn is_disjoint(first: &MarkerTree, second: &MarkerTree) -> bool {
             string_is_disjoint(expr1, expr2)
         }
         MarkerExpression::Extra { operator, name } => extra_is_disjoint(operator, name, expr2),
-        MarkerExpression::Arbitrary { .. } => false,
+        // `Arbitrary` expressions always evaluate to `false`, and are thus always disjoint.
+        MarkerExpression::Arbitrary { .. } => true,
```

Comment:
@ibraheemdev -- is this wrong? I was trying to understand the comment here: https://github.com/astral-sh/uv/pull/3679#issuecomment-2121392361

Reply:
This sounds right to me fwiw

Reply:
Yes, that is correct. However, I wonder if maybe we should be pessimistic regarding forking on meaningless expressions, cc @BurntSushi.

Also if we do make this change, I expected a test to be failing. I think we need to check if `expr2` is `Arbitrary` as well, I can make that change in a later PR.

Reply:
Okay, think I fixed this.

Reply:
I'm not fully sure. My instinct is also that we probably ought to be pessimistic on forking here. If we say two arbitrary expressions are disjoint, then we will fork. But logically speaking, if the arbitrary expressions are always false, then, well, I guess that makes sense?

I'm happy to let practice dictate what we do here. (So this is fine for now.)